Initial code + README commit

cat-haines opened the pull request.

cat-haines committed:
Initial code + README commit

cat-haines committed:
Minor updates in readme and class file
- Removed CLICK_OFF static which is ONLY used internally (and only once)
- Added square brackets around params for configureInitialInt method

cat-haines committed:
Renamed interrupt methods from ...Int to ...Interrupt

cat-haines committed:
Fixed interrupt names in README

cat-haines committed:
Renamed init to reset, and not calling from ctor

DoronShapiro commented on LIS3DH.class.nut:
```diff
@@ -0,0 +1,362 @@
+        local y_raw = (_getReg(OUT_Y_H) << 8) + _getReg(OUT_Y_L);
+        local z_raw = (_getReg(OUT_Z_H) << 8) + _getReg(OUT_Z_L);
+
+        //server.log(format("%02X, %02X, %02X",x_raw, y_raw, z_raw));
```
Commented debug statement

DoronShapiro commented on LIS3DH.class.nut:
```diff
@@ -0,0 +1,362 @@
+
+    // Enable/disable and configure inertial interrupts
+    function configureInertialInterrupt(state, threshold = 1, duration = 5, options = null) {
+        // Set default value for options (using statics, so can't set in )
```
Missing a word here?

DoronShapiro commented on README.md:
````diff
@@ -0,0 +1,286 @@
+accel.configureClickInt(true, LIS3DH.DOUBLE_CLICK);
+```
+
+### configureDataReadyInterrupt(*state*)
````
This a "configure" method, while `setInterruptLatching()` is a "set" method, but they both just enable/disable a feature.  We should probably try to be consistent with naming here.

DoronShapiro commented on LIS3DH.class.nut:
```diff
@@ -0,0 +1,362 @@
+        if (val != null) {
+            return val[0];
+        } else {
+            return null;
```
A `null` return type is never checked for when this method is used and may cause issues.

DoronShapiro commented on LIS3DH.class.nut:
```diff
@@ -0,0 +1,362 @@
+    }
+
+    function _setReg(reg, val) {
+        _i2c.write(_addr, format("%c%c", reg, (val & 0xff)));
```
Should we validate that I2C operations succeed?
(i.e. by checking the return value)

DoronShapiro commented on LIS3DH.class.nut:
```diff
@@ -0,0 +1,362 @@
+        _i2c.write(_addr, format("%c%c", reg, (val & 0xff)));
+    }
+
+    function _setRegBit(reg, bit, state) {
```
What does _state_ do?

cat-haines replied:
_setRegBit sets a specific bit in the register (so state is the state we're setting of that bit).

cat-haines committed:
Minor updates per PR #1
- _setReg and _getReg now throw errors on failure
- Inertial int now defaults to 2G instead of 1G
- Changed setInterruptLatching -> configureInterruptLatching
- Fixed some typos

cat-haines committed:
Removed second _twosComp method
